exporter/datadogexporter/integrationtest/integration_test.go

```diff
@@ -414,3 +438,89 @@ func sendTracesComputeTopLevelBySpanKind(t *testing.T) {
 	}
 	time.Sleep(1 * time.Second)
 }
+
+func TestIntegrationLogs(t *testing.T) {
+	// 1. Set up mock Datadog server
+	// See also https://github.com/DataDog/datadog-agent/blob/49c16e0d4deab396626238fa1d572b684475a53f/cmd/trace-agent/test/backend.go
+	seriesRec := &testutil.HTTPRequestRecorderWithChan{Pattern: testutil.MetricV2Endpoint, ReqChan: make(chan []byte)}
+	doneChannel := make(chan bool)
+	var connectivityCheck sync.Once
+	var logsData testutil.JSONLogs
+	server := testutil.DatadogLogServerMock(seriesRec.HandlerFunc, func() (string, http.HandlerFunc) {
+		return "/api/v2/logs", func(w http.ResponseWriter, r *http.Request) {
+			doneConnectivityCheck := false
+			connectivityCheck.Do(func() {
+				// The logs agent performs a connectivity check upon initialization.
+				// This function mocks a successful response for the first request received.
+				w.WriteHeader(http.StatusAccepted)
+				doneConnectivityCheck = true
+			})
+			if !doneConnectivityCheck {
+				jsonLogs := testutil.ProcessLogsAgentRequest(w, r)
+				logsData = append(logsData, jsonLogs...)
+				doneChannel <- true
+			}
+		}
+	})
+	defer server.Close()
+	t.Setenv("SERVER_URL", server.URL)
+
+	// 2. Start in-process collector
+	err := featuregate.GlobalRegistry().Set("exporter.datadogexporter.UseLogsAgentExporter", true)
+	assert.NoError(t, err)
+	factories := getIntegrationTestComponents(t)
+	app := getIntegrationTestCollector(t, "integration_test_logs_config.yaml", factories)
+	go func() {
+		assert.NoError(t, app.Run(context.Background()))
+	}()
+	defer app.Shutdown()
+
+	waitForReadiness(app)
+
+	// 3. Generate and send logs
+	sendLogs(t, 5)
+
+	// 4. Validate logs and metrics from the mock server
+	// Wait until `doneChannel` is closed and prometheus metrics are received.
+	var metricMap seriesMap
+	var metricMapStr string
+	for !strings.Contains(metricMapStr, "otelcol_receiver_accepted_log_records") && !strings.Contains(metricMapStr, "otelcol_exporter_sent_log_records") {
```

Comment:
Instead of doing this you can configure the prometheus receiver to only scrape these 2 metrics

Reply:
I tried this but there were still other metrics being sent (e.g. running metrics and scrape metrics). I added a condition in the test to filter out these metrics.